application/users/user.py

```diff
@@ -39,5 +51,5 @@ def update(self, **properties):
 
     def pre_save(self):
         super()
-        if self.does_not_exist():
+        if self.does_not_exist() and self.password:
```

Comment:
I'd appreciate it if we could use the `has_password` from further up for consistency sake here.